misc-utils/kill.c

```diff
@@ -551,10 +557,27 @@ static int kill_verbose(const struct kill_control *ctl)
 		rc = sigqueue(ctl->pid, ctl->numsig, ctl->sigdata);
 	else
 #endif
-		rc = kill(ctl->pid, ctl->numsig);
+#ifdef USE_KILL_WITH_PIDFDFS
+		if ((ctl->pidfd_ino > 0)) {
+			int pfd;
+			if ((pfd = pidfd_open(ctl->pid, 0)) < 0) {
```

Comment:
Don't do the assignment inside the conditional. Just split them into two statements.